aws/services/ecs/networking.tf

```
@@ -0,0 +1,60 @@
+data "aws_availability_zones" "available_zones" {
+  state = "available"
+}
+
+resource "aws_vpc" "test-ecs-vpc" {
+  cidr_block = "10.32.0.0/16"
+}
+
+resource "aws_subnet" "test-ecs-public" {
+  count                   = 2
+  availability_zone       = data.aws_availability_zones.available_zones.names[count.index]
+  cidr_block              = cidrsubnet(aws_vpc.test-ecs-vpc.cidr_block, 8, 2 + count.index)
+  vpc_id                  = aws_vpc.test-ecs-vpc.id
+  map_public_ip_on_launch = true
+}
+
+resource "aws_subnet" "test-ecs-private" {
+  count             = 2
+  availability_zone = data.aws_availability_zones.available_zones.names[count.index]
+  cidr_block        = cidrsubnet(aws_vpc.test-ecs-vpc.cidr_block, 8, count.index)
+  vpc_id            = aws_vpc.test-ecs-vpc.id
+}
+
+resource "aws_internet_gateway" "test-ecs-gateway" {
+  vpc_id = aws_vpc.test-ecs-vpc.id
+}
+
+resource "aws_route" "test-ecs-internet_access" {
+  route_table_id         = aws_vpc.test-ecs-vpc.main_route_table_id
+  destination_cidr_block = "0.0.0.0/0"
+  gateway_id             = aws_internet_gateway.test-ecs-gateway.id
+}
+
+resource "aws_eip" "test-ecs-gateway" {
+  count      = 2
+  vpc        = true
+  depends_on = [aws_internet_gateway.test-ecs-gateway]
+}
+
+resource "aws_nat_gateway" "test-ecs-gateway" {
```

if the [terraform naming conventions](https://www.terraform-best-practices.com/naming) they recomment to use `this` if there is no good name
```suggestion
resource "aws_nat_gateway" "this" {
```